concent_api/core/tests/test_integration_subtask_results_verify.py

```diff
@@ -196,6 +196,46 @@ def test_that_concent_responds_with_service_refused_when_request_arrives_too_lat
         )
         self._assert_stored_message_counter_not_increased()
 
+    def test_that_concent_responds_with_service_refused_when_request_arrives_too_early(self):
+        """
+        Provider -> Concent: SubtaskResultsVerify
+        Concent -> Provider: ServiceRefused (InvalidRequest)
+        """
+        # given
+        serialized_subtask_results_verify = self._get_serialized_subtask_results_verify(
+            subtask_results_verify=self._get_deserialized_subtask_results_verify(
+                timestamp="2018-04-01 10:30:00",
+                subtask_results_rejected=self._get_deserialized_subtask_results_rejected(
+                    reason=message.tasks.SubtaskResultsRejected.REASON.VerificationNegative,
+                    timestamp="2018-04-01 10:30:00",
+                    report_computed_task=self._get_deserialized_report_computed_task(
+                        timestamp="2018-04-01 10:01:00",
+                        subtask_id=self.subtask_id,
+                    ),
+                )
+            )
+        )
+
+        # when
+        with freeze_time("2018-04-01 10:29:59"):
+            response = self.client.post(
+                reverse('core:send'),
+                data=serialized_subtask_results_verify,
+                content_type='application/octet-stream',
+            )
+
+        # then
+        self._test_response(
+            response,
+            status=200,
+            key=self.PROVIDER_PRIVATE_KEY,
+            message_type=message.concents.ServiceRefused,
+            fields={
+                'reason': message.concents.ServiceRefused.REASON.InvalidRequest,
```

Comment:
Here we hava a different behaviour when request is too early... @cameel, shouldn't we have it alligned somehow among all Use Cases?

Reply:
What do you mean by 'different behaviour' ?

Reply:
sorry, I wasn't clear enough ;)
I mean that in some handlers, if the message is to early:
- we send back 'ServiceRefused' 
- rasie `Http400` (`handle_send_reject_report_computed_task`, `handle_send_ack_report_computed_task`,)
- `AcceptanceTimeLimitExceeded` (`handle_send_force_get_task_result`)
- `RequestPremature` (`handle_send_force_subtask_results`) 
- `ServiceRefused.REASON.InvalidRequest` (`handle_send_subtask_results_verify`)

It would be good to have it more unified ;)

Reply:
Right, but this is actually in golem messages. We are using what this module gives us. I agree that it can be unified but we can't do it on our own.

Reply:
@cameel , I would be for creating ticket(s) for Golem, what would you say?

Reply:
I agree with @dybi. We should ask golem to add new codes or make existing ones consistent and use them.

Also, we should strive to eliminate HTTP 400 responses in cases that are not tied to HTTP (invalid times, validation errors, etc.).